plutus-pab/src/Plutus/PAB/Core/ContractInstance.hs

```diff
@@ -94,10 +95,28 @@ activateContractSTM' ::
     -> (Eff appBackend ~> IO)
     -> ContractActivationArgs (ContractDef t)
     -> Eff effs ContractInstanceId
-activateContractSTM' ContractInstanceState{contractState,stmState} activeContractInstanceId runAppBackend a@ContractActivationArgs{caID, caWallet} = do
-  logDebug @(ContractInstanceMsg t) $ InitialisingContract caID activeContractInstanceId
+activateContractSTM' c@ContractInstanceState{contractState} activeContractInstanceId runAppBackend a@ContractActivationArgs{caID} = do
+  logInfo @(ContractInstanceMsg t) $ InitialisingContract caID activeContractInstanceId
   Contract.putStartInstance @t a activeContractInstanceId
   Contract.putState @t a activeContractInstanceId contractState
+  startContractInstanceThread' c activeContractInstanceId runAppBackend a
+
+-- | Spin up the STM Instance thread for the provided contract.
+startContractInstanceThread' ::
+    forall t m appBackend effs.
+    ( Member (LogMsg (ContractInstanceMsg t)) effs
+    , Member (Reader InstancesState) effs
+    , Contract.PABContract t
+    , AppBackendConstraints t m appBackend
+    , LastMember m (Reader ContractInstanceId ': appBackend)
+    , LastMember m effs
+    )
+    => ContractInstanceState t
+    -> ContractInstanceId
+    -> (Eff appBackend ~> IO)
+    -> ContractActivationArgs (ContractDef t)
+    -> Eff effs ContractInstanceId
+startContractInstanceThread' ContractInstanceState{stmState} activeContractInstanceId runAppBackend a@ContractActivationArgs{caID, caWallet} = do
```

Comment:
I split this out because, when the earlier two contract-store effects were being run for contracts _already_ in the database; it was blocking here. I think in particular the `putStartInstance` would crash silently? ( because the contractInstanceId is a primary key ).

I'll try and think of a better name for this function.

Reply:
@j-mueller I'm having a mental blank coming up with a better name; I've improved the comment. If inspiration strikes you let me know and I'll rename it!